### PR TITLE
BGDIINF_SB-1739, BGDIINF_SB-1764: SPEC corrections and added asset upload in SPEC

### PR DESCRIPTION
**Pull request by ltshb**
Added asset upload in spec.

Also did some corrections:

- Removed wrong openapi response in PATCH asset
- Moved SPEC generic response and schemas
- Removed the checksum:multihash from write Assets requests + clean ups
- Reorganized the requests by adding two new tags
- Moved the ETag definition from schemas to headers
- Made the collection links optional in write request

I suggest to do the review commit by commit, this will be easier to understands the changes. See also commit messages.

### CODE REVIEW

File: spec/static/spec/v0.9/openapitransactional.yaml [hunk 1]
```diff
@@ -21,21 +21,86 @@ tags:
...
   name: STAC
-- description: |
-    All write requests require authentication. The currently available options for a user to
-    authenticate himself are described below.
+- name: Data Management
+  description: |
+    Metadata management requests. Theses requests are used to create, update or delete the STAC
+    metadata.
+
+    *NOTE: these requests requires authentication as described in [here](#tag/Authentication).*
```
Comment (hansmannj) [inline spec/static/spec/v0.9/openapitransactional.yaml:29]: picky, I know: -s in `requires`

File: spec/static/spec/v0.9/openapitransactional.yaml [hunk 2]
```diff
@@ -21,21 +21,86 @@ tags:
...
+- name: Asset Upload Management
+  description: |
+    Asset file can be uploaded via the STAC API using the following requests.
+
+    *NOTE: the POST requests requires authentication as described in [here](#tag/Authentication).*
```
Comment (hansmannj) [inline spec/static/spec/v0.9/openapitransactional.yaml:34]: and here, too ;-)

File: spec/static/spec/v0.9/openapitransactional.yaml [hunk 3]
```diff
@@ -1950,27 +2004,318 @@ components:
...
       type: string
+    uploadId:
+      title: ID
+      type: string
+      description: Unique Asset upload identifier
+      example: KrFTuglD.N8ireqry_w3.oQqNwrYI7SfSXpVRiusKah0YigDnuM06hfJNIUZg4R_No0MMW9FLU2UG5anTW0boTUYVxKfBZWCFXqnQTpjnQEo1K7la39MYpjSTvIbZgnG
+      readOnly: true
+    dtUploadCreated:
+      title: created
+      description: Date time when the Asset's upload has been created/started.
```
Comment (hansmannj) [inline spec/static/spec/v0.9/openapitransactional.yaml:2021]: we often use `datetime` as one word, and here and elsewhere also as two separate words. Might make sense to be consistent here

File: spec/static/spec/v0.9/openapitransactional.yaml [hunk 4]
```diff
@@ -1950,27 +2004,318 @@ components:
...
+      title: CompleteUpload
+      type: object
+      required:
+      - parts
+      properties:
+        parts:
+          type: array
+          description: Parts that have been uploaded
+          items:
+            title: File part that have been uploaded
```
Comment (hansmannj) [inline spec/static/spec/v0.9/openapitransactional.yaml:2131]: again a picky one, sorry: +s in for `part` here ;-)

**Review by hansmannj — APPROVED** [review]
:1st_place_medal: 
LGTM, thanks!

File: spec/static/spec/v0.9/openapitransactional.yaml (hunk 1 above)
Comment (ltshb) [reply spec/static/spec/v0.9/openapitransactional.yaml:29]: Not picky, correctness :-P I'm glad you find my bad english though thanks.

File: spec/static/spec/v0.9/openapitransactional.yaml (hunk 3 above)
Comment (ltshb) [reply spec/static/spec/v0.9/openapitransactional.yaml:2021]: Well here it is a description where we refer to date and time, I'll add `and`. When we refer to `datetime`, it should be a reference to the `datetime` type. Same same but different :-)

**Pull request merged**